utils/decision.py

```diff
@@ -0,0 +1,60 @@
+from enum import Enum, auto
+
+from utils.result import ResultTypes
+
+
+class DecisionStrategies(Enum):
```

Comment:
each decision should have documentation saying what it does. either in their funky impls or the enum values, but e.g. for strict: 'computes the strict decision' doesnt explain what it does

Reply:
also shouldnt this name be singular, `DecisionStrategy`?